src/frontends/pytorch/src/op/atan2.cpp

```diff
@@ -0,0 +1,101 @@
+// Copyright (C) 2018-2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#define _USE_MATH_DEFINES
+
+#include <math.h>
+
+#include <memory>
+
+#include "openvino/core/type/element_type.hpp"
+#include "openvino/frontend/pytorch/node_context.hpp"
+#include "openvino/op/add.hpp"
+#include "openvino/op/atan.hpp"
+#include "openvino/op/constant.hpp"
+#include "openvino/op/convert_like.hpp"
+#include "openvino/op/divide.hpp"
+#include "openvino/op/equal.hpp"
+#include "openvino/op/greater.hpp"
+#include "openvino/op/greater_eq.hpp"
+#include "openvino/op/less.hpp"
+#include "openvino/op/logical_and.hpp"
+#include "openvino/op/logical_or.hpp"
+#include "openvino/op/multiply.hpp"
+#include "openvino/op/select.hpp"
+#include "openvino/op/subtract.hpp"
+#include "utils.hpp"
+
+namespace ov {
+namespace frontend {
+namespace pytorch {
+namespace op {
+
+using namespace ov::op;
+
+OutputVector translate_atan2(const NodeContext& context) {
+    // atan2(input, other, *) → Tensor
+    num_inputs_check(context, 2, 2);
+    Output<Node> lhs;
+    Output<Node> rhs;
+
+    std::tie(lhs, rhs) = get_inputs_with_promoted_types(context, 0, 1);
+
+    auto div = context.mark_node(std::make_shared<v1::Divide>(lhs, rhs));
+
+    auto atan = context.mark_node(std::make_shared<v0::Atan>(div));
+
+    // create some constants to adjust result according to quadrant.
+    auto zero = context.mark_node(v0::Constant::create(ov::element::i32, Shape{}, {0}));
+    auto pi = context.mark_node(v0::Constant::create(ov::element::f64, Shape{}, {M_PI}));
+    auto neg_pi = context.mark_node(v0::Constant::create(ov::element::f64, Shape{}, {-M_PI}));
+    auto half_pi = context.mark_node(v0::Constant::create(ov::element::f64, Shape{}, {M_PI_2}));
+    auto neg_half_pi = context.mark_node(v0::Constant::create(ov::element::f64, Shape{}, {-M_PI_2}));
+
+    zero = context.mark_node(std::make_shared<v1::ConvertLike>(zero, rhs));
+    pi = context.mark_node(std::make_shared<v1::ConvertLike>(pi, rhs));
+    neg_pi = context.mark_node(std::make_shared<v1::ConvertLike>(neg_pi, rhs));
```

Comment:
```suggestion
    auto half_pi = context.mark_node(v0::Constant::create(ov::element::f64, Shape{}, {M_PI_2}));
    auto neg_half_pi = context.mark_node(v0::Constant::create(ov::element::f64, Shape{}, {-M_PI_2}));

    zero = context.mark_node(std::make_shared<v1::ConvertLike>(zero, rhs));
    pi = context.mark_node(std::make_shared<v1::ConvertLike>(pi, rhs));
```
You are not using `neg_pi` in your code.